src/partisan_hiscamp_membership_strategy.erl

```diff
@@ -0,0 +1,251 @@
+    MembershipList = membership_list(State),
+    {ok, MembershipList, State}.
+
+%% remote node connects, notify node to us. perform forwarding?
+%% Membership0 is a set of node elements
+join(#hiScamp{membership=Membership0, level1=L1, level2=L2}=State0, Node, _NodeState) ->
+    OutgoingMessages0 = [],
+    case partisan_config:get(tracing, ?TRACING) of
+        true ->
+            lager:info("~p: Adding node ~p to our membership.", [node90, Node]);
+        false ->
+            ok
+    end,
+    Membership = sets:add_element(Node, Membership0),
+%% notify node to add us to its state.
+    Myself = partisan_peer_service_manager:myself(),
+    OutgoingMessages1 = OutgoingMessages0 ++ [{Node, {protocol, {forward_subscription, Myself}}}],
+    OutgoingMessages2 = sets:fold(fun(N, OM) ->
+        %% tracing is a macro
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding subscription for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+            end,
+            OM ++ [{N, {protocol, {forward_subscription, Node}}}]
+            end, OutgoingMessages1, Membership0),
+    C = partisan_config:get(scamp_c, ?SCAMP_C_VALUE),
+    ForwardMessages = lists:map(fun(N) ->
+        case partisan_config:get(tracing, ?TRACING) of
+            true ->
+                lager:info("~p: Forwarding additional subs. for ~p to node: ~p", [node(), Node, N]);
+            false ->
+                ok
+        end,
+        {N, {protocol, {forward_subscription, Node}}}
+        end, select_random_sublist(state0, C)),
+    OutgoingMessages = OutgoingMessages2 ++ ForwardMessages,
+    State = State0#hiScamp{membership=Membership},
+    MembershipList = membership_list(State),
+    {ok, Membership, OutgoingMessages, State}.
+
+distance(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2}=_State0, Node) ->
+    % A node j joins the system by sending a subscription request to the node s which is closest to it
+    Threshold = 3,
+    Dict = get(distance_metrics),
+    Dist = dict:find(Node, Dict),
+    % handle errors
+    case Threshold > Dist of
+        true -> 
+            % process this case using Scamp within this cluster
+            % subscription at level 1
+            %need to send copies out to other nodes -> gossip protocol
+            sets:add(Node, L1);
+        false ->
+                       % search through set to find another set
+                       % store two sets in the record
+            % subscription at level 2
+                       %input Cluster0
+            sets:add(Node, L2)
+    end,
+    heaps:add(Dist, Heap).
+
+get_centroid_two_clusters(#hiScamp{membership=Membership0, heap=Heap, level1=L1, level2=L2} =State0) ->
+    Min1 = heaps:min(Heap),
+    Heap = heaps:delete_min(Heap),
+    Min2 = heaps:min(Heap),
+    case sets:is_element(Min1, L1) andalso sets:is_element(Min2, L1) of
+        ok
```

Comment:
Not sure why this is here, this is a syntax error.  You're in a case, and you need to pattern match.